### PR TITLE
core: rename api for better semantic

### PR DESCRIPTION
**Pull request by FeelyChau**
Pipeline:
* dataSource -> datasource
* artifacts -> artifact
* specVersion -> version  ??

Runtime:
* dataSource -> dataset

### CODE REVIEW

File: docs/rfcs/0000-new-pipeline.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Introducing a new Pipeline syntax for supporting:
 
 ```json
 {
-  "specVersion": "2.0",
+  "version": "2.0",
```
Comment (yorkie) [inline docs/rfcs/0000-new-pipeline.md:18]: The `version` is reversed for a pipeline itself.

**Comment by yorkie** [comment]
And I'm -1 on renaming datasource to the dataset, the data source actually represents a data schema for multiple datasets.

File: docs/rfcs/0000-new-pipeline.md (hunk 1 above)
Comment (FeelyChau) [reply docs/rfcs/0000-new-pipeline.md]: OK, rollbacked.

**Comment by FeelyChau** [comment]
> And I'm -1 on renaming datasource to the dataset, the data source actually represents a data schema for multiple datasets.

If you mean the datasource type will be changed, I think we should confirm it before 2.0 releasing. Otherwise, if the interface type is exactly equal to `Dataset`, I think the name `dataset` is better, model script should not care the `dataflow`.

**Comment by yorkie** [comment]
Dataset is the concept of datacook, the data source is Pipcook's.

**Comment by FeelyChau** [comment]
> Dataset is the concept of datacook, the data source is Pipcook's.

What's the different except the name? In my opinion, they are the same.

**Comment by yorkie** [comment]
Good question @FeelyChau, and I searched some for it:

- https://community.powerbi.com/t5/Service/Difference-between-data-source-and-data-set/m-p/531191
- https://www.sqlservercentral.com/forums/topic/what-is-the-relationship-between-datasources-and-datasets

In summary, they are not the same thing, and Data Source is a place where we fetch data or dataset, and the Dataset is a set of data, which represents the data itself. Therefore I'm agreed with using "dataset" :)

**Review by yorkie — APPROVED** [review]

**Pull request merged**